python/cudf/cudf/io/csv.py

```diff
@@ -18,45 +18,46 @@
 @ioutils.doc_read_csv()
 def read_csv(
     filepath_or_buffer,
-    lineterminator="\n",
-    quotechar='"',
-    quoting=0,
-    doublequote=True,
-    header="infer",
-    mangle_dupe_cols=True,
-    usecols=None,
     sep=",",
     delimiter=None,
-    delim_whitespace=False,
-    skipinitialspace=False,
+    header="infer",
     names=None,
+    index_col=None,
+    usecols=None,
+    prefix=None,
+    mangle_dupe_cols=True,
     dtype=None,
-    skipfooter=0,
-    skiprows=0,
-    dayfirst=False,
-    compression="infer",
-    thousands=None,
-    decimal=".",
     true_values=None,
     false_values=None,
+    skipinitialspace=False,
+    skiprows=0,
+    skipfooter=0,
     nrows=None,
-    byte_range=None,
-    skip_blank_lines=True,
-    parse_dates=None,
-    comment=None,
     na_values=None,
     keep_default_na=True,
     na_filter=True,
-    prefix=None,
-    index_col=None,
+    skip_blank_lines=True,
+    parse_dates=None,
+    dayfirst=False,
+    compression="infer",
+    thousands=None,
+    decimal=".",
+    lineterminator="\n",
+    quotechar='"',
+    quoting=0,
+    doublequote=True,
+    comment=None,
+    delim_whitespace=False,
+    byte_range=None,
     use_python_file_object=True,
-    **kwargs,
+    storage_options=None,
+    bytes_per_thread=256_000_000,
```

Comment:
@rjzamora Could you help me understand about `bytes_per_thread` that you have added in https://github.com/rapidsai/cudf/pull/9265/, asking so that we can document this parameter in the docstring.

Reply:
`bytes_per_thread` is a poorly-named option for the [_fsspec_data_transfer](https://github.com/rapidsai/cudf/blob/cd60462f6c05e717352f32077e9d1f3a225271e6/python/cudf/cudf/utils/ioutils.py#L1664) utility, which is **only** used when `use_python_file_object` is set to `False`.

When  `use_python_file_object=False`, we are not able to perform partial IO from a remote file, so the `_fsspec_data_transfer` code simply transfers the entire file into local memory.  Since the file can be quite large, we get slightly better throughput by decomposing it and transfering multiple "blocks" in parallel (using a python thread pool). `<file-size-bytes> / bytes_per_thread` effectively tells us how many threads to use to transfer the file into local host memory.

Reply:
Thanks @rjzamora ! I updated the docstring.